mParticle-Apple-SDK/Data Model/MPForwardRecord.m

```diff
@@ -112,19 +112,22 @@ - (instancetype)initWithMessageType:(MPMessageType)messageType execStatus:(MPKit
     if (kitFilter.appliedProjections.count > 0) {
         NSMutableArray *projections = [[NSMutableArray alloc] initWithCapacity:kitFilter.appliedProjections.count];
         NSMutableDictionary *projectionDictionary;
+        NSString *currentProjectionName = ((MPEvent *)kitFilter.originalEvent).name;
         
         for (MPEventProjection *eventProjection in kitFilter.appliedProjections) {
-            projectionDictionary = [[NSMutableDictionary alloc] initWithCapacity:4];
-            projectionDictionary[kMPFRProjectionId] = @(eventProjection.projectionId);
-            projectionDictionary[kMPMessageTypeKey] = NSStringFromMessageType(messageType);
-            
-            projectionDictionary[kMPEventTypeKey] = NSStringFromEventType(eventProjection.eventType);
-            
-            if (eventProjection.projectedName) {
-                projectionDictionary[kMPFRProjectionName] = eventProjection.projectedName;
+            if (eventProjection.projectedName == currentProjectionName) {
```

Comment:
In Objective-C, you can't use `==` to compare objects (including strings). The `==` will just compare if the pointers are equal, there's no operator overloading. So you need to compare using `isEqual:`
```suggestion
            if ([eventProjection.projectedName isEqual:currentProjectionName]) {
```

Reply:
jus added this change